File: Client/src/app/models/CookedRecipeDTO.ts
import { CookedRecipeCalledIngredientDetailsDTO } from "./CookedRecipeCalledIngredientDetailsDTO";
import { RecipeDTO } from "./RecipeDTO";

export class CookedRecipeDTO {
  id?: number = undefined;
  recipe?: RecipeDTO = undefined;
  recipeId?: number = undefined;
  cookedRecipeCalledIngredients?: CookedRecipeCalledIngredientDetailsDTO[] = undefined;
}

File: API/ContainerNinja.Contracts/DTO/CookedRecipeDTO.cs
using ContainerNinja.Contracts.Data.Entities;

namespace ContainerNinja.Contracts.DTO
{
    public class CookedRecipeDTO
    {
        public int Id { get; set; }

        public RecipeDTO Recipe { get; set; }

        public int RecipeId { get; set; }

        public IList<CookedRecipeCalledIngredientDTO> CookedRecipeCalledIngredients { get; set; }
    }
}